### PR TITLE
Make the "gray" state clearer

### PR DESCRIPTION
**Pull request by mgurgel**
**Reviewer:** @shakyShane 

Asana task: https://app.asana.com/0/1206594217596623/1207737672183274/f

Preview: https://deploy-preview-164--glittering-madeleine-71fec0.netlify.app/html/iframe
- [Updated Gray State](https://deploy-preview-164--glittering-madeleine-71fec0.netlify.app/html/android.html?screen=primaryScreen&state=protectionsOn_allowedTrackers)
- [New Green State: allowedFirstParty](https://deploy-preview-164--glittering-madeleine-71fec0.netlify.app/html/android.html?screen=primaryScreen&state=protectionsOn_allowedFirstParty)
- [New Green State: allowedFirstParty+allowedNonTrackers](https://deploy-preview-164--glittering-madeleine-71fec0.netlify.app/html/android.html?screen=primaryScreen&state=protectionsOn_allowedFirstParty_allowedNonTrackers)

## Description:

Make changes to the logic and copy of the privacy dashboard's "gray" state to make it clearer.

## Steps to test this PR:

**Option 1 - Preview env**
- Check out this branch
- Run `npm run preview`
- Check the following states:
  - _protectionsOn_allowedTrackers_ - Updated Gray State
  - _protectionsOn_allowedTrackers_allowedNonTrackers_ - Updated Gray State
  - _protectionsOn_allowedFirstParty_ - New Green State, First Party Requests only
  - _protectionsOn_allowedFirstParty_allowedNonTrackers_ - New Green State, First Party and Non-Blocking Third Party Requests

**Option 2 - Development env in browser**
- Check out branch [mgurgel/privacy-dashboard-new-green-state](https://github.com/duckduckgo/macos-browser/tree/mgurgel/privacy-dashboard-new-green-state) of the macOS Browser
- Run browser in Xcode
- Navigate to sites such as ebay.de (new green state) and reddit.com (updated gray state) and toggle Privacy Dashboard

### CODE REVIEW

File: integration-tests/macos.spec-int.js [hunk 1]
```diff
@@ -43,6 +43,14 @@ test('insecure certificate', async ({ page }) => {
     await dash.hasInsecureTextDetail()
 })
 
+test('allowed first party requests', async ({ page }) => {
```
Comment (mgurgel) [inline integration-tests/macos.spec-int.js:46]: I added a test case for the the new state, just for added confidence beyond the screenshots.

File: shared/js/ui/templates/key-insights.js [hunk 2]
```diff
@@ -31,7 +32,11 @@ export function renderKeyInsight(modelOverride) {
         if (model.isBroken) return keyInsightsState.broken
         if (!model.protectionsEnabled) return keyInsightsState.userAllowListed
         if (model.isaMajorTrackingNetwork && model.tab.parentEntity) return keyInsightsState.majorTrackingNetwork
+        // TODO: Can we refactor this?
         if (model.tab.requestDetails.blocked.requestCount === 0) {
```
Comment (mgurgel) [inline shared/js/ui/templates/key-insights.js:36]: Gonna leave the refactor of this bit for the near future if that’s ok

File: integration-tests/DashboardPage.js [hunk 3]
```diff
@@ -83,10 +84,12 @@ export class DashboardPage {
         await this.showsConnectionScreen()
         await this.screenshot(name + '-state-connection.png', opts)
         await this.goBack()
-        await this.viewTrackerCompanies()
-        await this.showsTrackersScreen()
-        await this.screenshot(name + '-state-trackers.png', opts)
-        await this.goBack()
+        if (this.shouldScreenshotTrackersScreen(state)) {
```
[the remark this answers is not in the document]
Comment (shakyShane) [reply integration-tests/DashboardPage.js:87]: I would focus the language around what the user would experience, like 'is the link there?'

File: integration-tests/DashboardPage.js [hunk 4]
```diff
@@ -97,6 +100,21 @@ export class DashboardPage {
...
+        switch (state) {
+            case testDataStates['fire-button']:
+            case testDataStates.protectionsOn_allowedFirstParty:
+            case testDataStates.protectionsOn_allowedFirstParty_allowedNonTrackers:
+                return false
+            default:
+                return true
+        }
+    }
+
```
Comment (shakyShane) [inline integration-tests/DashboardPage.js:117]:
```suggestion
    /**
     * Whether to screenshot the company trackers screen, as some states omit that screen
     */
    async shouldScreenshotTrackersScreen() {
        let count = await this.trackerCompaniesLink().count()
        return count === 1
    }

```

File: integration-tests/DashboardPage.js (hunk 3 above)
Comment (shakyShane) [inline integration-tests/DashboardPage.js:87]: I would just look if the link exists first:

```suggestion
        if (await this.shouldScreenshotTrackersScreen()) {
```

**Review by shakyShane — CHANGES_REQUESTED** [review]
Excellent work - just the test to tweak.

File: integration-tests/DashboardPage.js (hunk 4 above)
Comment (mgurgel) [reply integration-tests/DashboardPage.js]: Changed. Thanks!

**Review by shakyShane — APPROVED** [review]

**Pull request merged**